Tests/JWEHeaderTests.swift

```diff
@@ -51,23 +51,23 @@ class JWEHeaderTests: XCTestCase {
 
         XCTAssertEqual(header.parameters["enc"] as? String, SymmetricKeyAlgorithm.A256CBCHS512.rawValue)
         XCTAssertEqual(header.parameters["alg"] as? String, AsymmetricKeyAlgorithm.RSA1_5.rawValue)
-        XCTAssertEqual(header.data(), try! JSONSerialization.data(withJSONObject: parameterDictRSA, options: []))
+        XCTAssertEqual(header.data().count, try! JSONSerialization.data(withJSONObject: parameterDictRSA, options: []).count)
```

Comment:
Why switch to comparing lengths instead of comparing data directly? This seems like a downgrade in terms of test guarantees.

Reply:
Sorry, I should have pointed out the reason for this change or moved it to another pr.

Comparing the data is flaky since we cannot guarantee the order of the keys in the (JSON) data. Previously the tests used to fail sometimes if we were unlucky and the keys were in different orders. I think I've come up with a better solution for it now.

Reply:
Aha, thanks for the explanation. In that case I suppose it's reasonable.